src/component/admin/auth/simple-auth.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState, useContext, useEffect } from 'react';
+import {
+    Button,
+    FormControlLabel,
+    Grid,
+    Switch,
+} from '@material-ui/core';
+import { Alert } from '@material-ui/lab';
+import PageContent from '../../common/PageContent/PageContent';
+import AccessContext from '../../../contexts/AccessContext';
+import { ADMIN } from '../../providers/AccessProvider/permissions';
+import useAuthSettings from '../../../hooks/api/getters/useAuthSettings/useAuthSettings';
+import useAuthSettingsApi, {ISimpleAuthSettings } from '../../../hooks/api/actions/useAuthSettingsApi/useAuthSettingsApi';
+
+function SimpleAuth() {
+
+    const { config } = useAuthSettings('simple');
+    const [disabled, setDisabled] = useState<boolean>(false);
+    const [info, setInfo] = useState<string>();
+    const { updateSettings, errors } = useAuthSettingsApi<ISimpleAuthSettings>('simple')
+    const { hasAccess } = useContext(AccessContext);
+
+
+    useEffect(() => {
+        setDisabled(!!config.disabled);
+    }, [ config.disabled ]);
+
+    if (!hasAccess(ADMIN)) {
+        return (
+            <Alert severity="error">
+                You need to be a root admin to access this section.
+            </Alert>
+        );
+    }
+
+    const updateDisabled = () => {
+        setDisabled(!disabled);
+    };
+
+    
+    const onSubmit = async e => {
+        e.preventDefault();
+        
+        setInfo('...saving');
+        try {
+            const settings: ISimpleAuthSettings = { disabled };
+            await updateSettings(settings);
+            setInfo('Settings stored');
+            setTimeout(() => setInfo(''), 2000);
+        } catch (e) {
+            setInfo('');
+            setDisabled(config.disabled)
+        }
+        
+    };
+    return (
+        <PageContent headerContent=''>
+            <form onSubmit={onSubmit}>
+                <Grid container spacing={3}>
+                    <Grid item md={5}>
+                        <strong>Password based login</strong>
+                        <p>Allow users to login with username & password</p>
+                    </Grid>
+                    <Grid item md={6} style={{ padding: '20px' }}>
+                        <FormControlLabel
+                            control={
+                                <Switch
+                                    onChange={updateDisabled}
+                                    value={!disabled}
+                                    name="disabled"
+                                    checked={!disabled}
```

Comment:
I know this one looks a bit strange, but it is on purpose. We want to store the value as "I have actively disabled password based login", and thus call the config option "disabled". 

But presenting it as disabled to the user can be a bit confusing, so we invert it to "Is password based authentication enabled".

Reply:
Yeah. I prefer when I look at the component to understand what is going on immediately. Disabled has a strong meaning in the web development world, that it's difficult for me to associate it with something else. I just know I will look at this component at some point and wonder why we are setting disabled as the value. Perhaps we could call it `[disablePasswordAuth, setDisablePasswordAuth]` ?

Reply:
thanks, make sense!